cassiopeia/solver/SpectralGreedySolver.py

```diff
@@ -50,33 +56,54 @@ class SpectralGreedySolver(GreedySolver.GreedySolver):
         priors: Prior probabilities of character state transitions
         tree: The tree built by `self.solve()`. None if `solve` has not been
             called yet
-        prune_cm: A character matrix with duplicate rows filtered out
-        weights: A set of optional weights for calculating similarity for edges
-            in the graph
+        unique_character_matrix: A character matrix with duplicate rows filtered
+            out
+        similarity_function: A function that calculates a similarity score
+            between two given samples and their observed mutations
+        weights: Weights on character/mutation pairs, derived from priors
+        threshold: A minimum similarity threshold
     """
 
     def __init__(
         self,
         character_matrix: pd.DataFrame,
-        missing_char: str,
+        missing_char: int,
         missing_data_classifier: Union[Callable, str] = "average",
         meta_data: Optional[pd.DataFrame] = None,
-        priors: Optional[Dict] = None,
+        priors: Optional[Dict[int, Dict[int, float]]] = None,
+        prior_function: Optional[Callable[[float], float]] = None,
+        similarity_function: Optional[
+            Callable[
+                [
+                    List[int],
+                    List[int],
+                    int,
+                    Optional[Dict[int, Dict[int, float]]],
+                ],
+                float,
+            ]
+        ] = None,
         threshold: Optional[int] = 0,
-        weights: Optional[Dict] = None,
     ):
 
-        super().__init__(character_matrix, missing_char, meta_data, priors)
+        super().__init__(
+            character_matrix, missing_char, meta_data, priors, prior_function
+        )
 
         self.missing_data_classifier = missing_data_classifier
         self.threshold = threshold
-        self.weights = weights
+        if similarity_function:
+            self.similarity_function = similarity_function
+        else:
+            self.similarity_function = (
+                dissimilarity_functions.hamming_similarity_without_missing
```

Comment:
If we default to `hamming_distance_without_missing`, can you make this clear in the documentation of the class?

Reply:
I think this is a working default for now, but I'd like to benchmark different similarities

Reply:
Can you put it in the documentation for now, and leave a TODO for experimenting to find the best default dissimilarity function?

Reply:
Yes will do